database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -140,18 +156,26 @@ public void onCancelled(DatabaseError error) {
         return mDataSnapshots;
     }
 
-    private int getIndexForKey(String key) {
-        int dataCount = size();
-        int index = 0;
-        for (int keyIndex = 0; index < dataCount; keyIndex++) {
-            String superKey = mKeySnapshots.getObject(keyIndex);
-            if (key.equals(superKey)) {
-                break;
-            } else if (mDataSnapshots.get(index).getKey().equals(superKey)) {
-                index++;
+    private int returnOrFindIndexForKey(int index, String key) {
+        int realIndex;
+        if (isKeyAtIndex(key, index)) {
+            realIndex = index;
+        } else {
+            int dataCount = size();
+            int dataIndex = 0;
+            for (int keyIndex = 0; dataIndex < dataCount; keyIndex++) {
+                if (keyIndex == mKeySnapshots.size()) { break; }
```

Comment:
I added this line to the old method to handle being called by a deletion at the last index.